File: solutions/2026-06-22/solution.py
# Valid Palindrome
# Difficulty: Easy
# URL: https://leetcode.com/problems/valid-palindrome/

def validPalindrome(s: str) -> bool:
  def is_palindrome(s: str) -> bool:
    return s == s[::-1]
  return is_palindrome(s) or (is_palindrome(s[:-1]) and is_palindrome(s[1:]))

File: solutions/2026-06-22/solution.java
// Valid Palindrome
// Difficulty: Easy
// URL: https://leetcode.com/problems/valid-palindrome/

class Solution {
    public boolean validPalindrome(String s) {
        return isPalindrome(s) || (isPalindrome(s.substring(0, s.length() - 1)) && isPalindrome(s.substring(1)));
    }
    public boolean isPalindrome(String s) {
        return s.equals(new StringBuilder(s).reverse().toString());
    }
}